app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -803,15 +771,23 @@ public void failTargetMismatch() {
 
     @Override
     public CustomProgressDialog generateProgressDialog(int taskId) {
-        if (taskId != DIALOG_INSTALL_PROGRESS) {
-            Log.w(TAG, "taskId passed to generateProgressDialog does not match "
-                    + "any valid possibilities in CommCareSetupActivity");
-            return null;
-        }
-        if (isSingleAppBuild()) {
-            return ConsumerAppsUtil.getGenericConsumerAppsProgressDialog(taskId, true);
-        } else {
-            return generateNormalInstallDialog(taskId);
+        switch (taskId) {
+            case DIALOG_INSTALL_PROGRESS:
+                if (isSingleAppBuild()) {
+                    return ConsumerAppsUtil.getGenericConsumerAppsProgressDialog(taskId, true);
+                } else {
+                    return generateNormalInstallDialog(taskId);
+                }
+            case ConnectionDiagnosticTask.CONNECTION_ID:
+                String title = Localization.get("connection.test.run.title");

```

Comment:
Nitpick: These strings made sense when user was explicity running a connection test by going into settings, but not so much when these connection test are part of another process. I would add new strings for this use case - 
title - "Checking Your Network Connection..."  , message - "We are checking your network connection to be able to proceed with installation"